experiments/PlaywrightSharp/PlaywrightSharp/ApiSurface/IBrowser.cs

```diff
@@ -0,0 +1,224 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *
+ * ------------------------------------------------------------------------------ 
+ * <auto-generated> 
+ * This code was generated by a tool at:
+ * /utils/doclint/generateDotnetApi.js
+ * 
+ * Changes to this file may cause incorrect behavior and will be lost if 
+ * the code is regenerated. 
+ * </auto-generated> 
+ * ------------------------------------------------------------------------------
+ */
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.IO;
+using System.Runtime.Serialization;
+using System.Text.Json;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace PlaywrightSharp
+{
+    /// <summary>
+	/// <para>
+	/// A Browser is created via <see cref="IBrowserType.LaunchAsync"/>. An example of using
+	/// a <see cref="IBrowser"/> to create a <see cref="IPage"/>:
+	/// </para>
+	/// </summary>
+	public partial interface IBrowser
+	{
+		/// <summary>
+		/// <para>
+		/// Emitted when Browser gets disconnected from the browser application. This might
+		/// happen because of one of the following:
+		/// </para>
+		/// <list type="bullet">
+		/// <item><description>Browser application is closed or crashed.</description></item>
+		/// <item><description>The <see cref="IBrowser.CloseAsync"/> method was called.</description></item>
+		/// </list>
+		/// </summary>
+		event EventHandler<IBrowser> Disconnected;
+	
+		/// <summary>
+		/// <para>
+		/// In case this browser is obtained using <see cref="IBrowserType.LaunchAsync"/>, closes
+		/// the browser and all of its pages (if any were opened).
+		/// </para>
+		/// <para>
+		/// In case this browser is connected to, clears all created contexts belonging to this
+		/// browser and disconnects from the browser server.
+		/// </para>
+		/// <para>
+		/// The <see cref="IBrowser"/> object itself is considered to be disposed and cannot
+		/// be used anymore.
+		/// </para>
+		/// </summary>
+		Task CloseAsync();
+	
+		/// <summary>
+		/// <para>
+		/// Returns an array of all open browser contexts. In a newly created browser, this
+		/// will return zero browser contexts.
+		/// </para>
+		/// </summary>
+		dynamic GetContexts();
```

Comment:
This could have a better return type
![image](https://user-images.githubusercontent.com/2198466/108055365-a42ab180-702e-11eb-94b7-b3af91e3255b.png)